packages/terra-overlay/src/Overlay.jsx

```diff
@@ -0,0 +1,176 @@
+    this.overflow = document.documentElement.style.overflow;
+
+    if (this.props.isRelativeToContainer) {
+      this.container = node.parentNode;
+      this.containerChildren = this.container.children;
+      this.disableContainerChildrenFocus();
+    } else {
+      document.documentElement.style.overflow = 'hidden';
+      this.container = null;
+    }
+  }
+
+  disableContainerChildrenFocus() {
+    if (this.containerChildren) {
+      const prevTabIndex = [];
+      for (let i = 0; i < this.containerChildren.length; i += 1) {
+        prevTabIndex.push(this.containerChildren[i].tabIndex);
+        this.containerChildren[i].tabIndex = -1;
+      }
+      this.containerChildrenPrevTabIndex = prevTabIndex;
+    }
+  }
+
+  enableContainerChildrenFocus() {
+    if (this.containerChildren) {
+      for (let i = 0; i < this.containerChildren.length; i += 1) {
+        this.containerChildren[i].tabIndex = this.containerChildrenPrevTabIndex[i];
+      }
+    }
+  }
+
+  shouldHandleESCKeydown(event) {
+    if (this.props.isOpen && event.keyCode === KEYCODES.ESCAPE) {
+      this.handleCloseEvent(event);
+      event.preventDefault();
+    }
+  }
+
+  shouldHandleClick(event) {
+    if (this.props.isOpen) {
+      this.handleCloseEvent(event);
+    }
+  }
+
+  handleCloseEvent(event) {
+    if (this.props.onRequestClose) {
+      this.props.onRequestClose(event);
+    }
+  }
+
+  resetBeforeOverlay() {
+    document.documentElement.style.overflow = this.overflow;
+    if (this.props.isRelativeToContainer) {
+      this.enableContainerChildrenFocus();
+    }
+  }
+
+  render() {
+    const { children, isOpen, backgroundStyle, isScrollable, isRelativeToContainer, onRequestClose, ...customProps } = this.props;
+    const type = isRelativeToContainer ? 'container' : 'fullscreen';
+
+    if (!isOpen) {
+      this.resetBeforeOverlay();
+      return null;
+    }
+
+    const OverlayClassNames = classNames([
+      'terra-Overlay',
+      { [`terra-Overlay--${type}`]: type },
+      { [`terra-Overlay--${backgroundStyle}`]: backgroundStyle },
+      { 'terra-Overlay--scrollable': isScrollable },
+      customProps.className,
+    ]);
+
+    // Disable linter to pass onClick to div element.
+    /* eslint-disable jsx-a11y/no-static-element-interactions */
+    const overlayComponent = (
+      <div {...customProps} ref={this.setContainer} onClick={this.shouldHandleClick} className={OverlayClassNames} tabIndex="0" >
+        <div className="terra-Overlay-content">
+          {children}
+        </div>
+      </div>
+    );
+    /* eslint-enable jsx-a11y/no-static-element-interactions */
+
+    if (isRelativeToContainer) {
+      return overlayComponent;
+    }
+
+    return (
+      <FocusTrap>
+        {overlayComponent}
+      </FocusTrap>
+    );
+  }
+}
+
+Overlay.propTypes = propTypes;
+Overlay.defaultProps = defaultProps;
+Overlay.BackgroundStyles = BackgroundStyles;
```

Comment:
nit: Update this from `Overlay.BackgroundStyles = BackgroundStyles;` to `Overlay.Opts.BackgroundStyles = BackgroundStyles;`